collector/ipvs.go

```diff
@@ -0,0 +1,258 @@
+// +build !noipvs
+
+package collector
+
+import (
+	"bufio"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	procNetIpvs      = "/proc/net/ip_vs"
+	procNetIpvsStats = "/proc/net/ip_vs_stats"
+	ipvsSubsystem    = "ipvs"
+)
+
+var (
+	ipvsStatFieldNames = []string{
+		"connections_total",
+		"incoming_packets_total",
+		"outgoing_packets_total",
+		"incoming_bytes_total",
+		"outgoing_bytes_total",
+	}
+	ipvsBackendLabelNames = []string{
+		"local_address",
+		"local_port",
+		"remote_address",
+		"remote_port",
+		"proto",
+	}
+)
+
+type ipvsCollector struct {
+	config                                                           Config
+	backendConnectionsActive, backendConnectionsInAct, backendWeight *prometheus.GaugeVec
+	stats                                                            map[string]prometheus.Counter
+}
+
+type ipvsBackendStatus struct {
+	localAddress  net.IP
+	localPort     uint16
+	proto         string
+	remoteAddress net.IP
+	remotePort    uint16
+	activeConn    uint16
```

Comment:
hmm the problem with that is that I don't know what to put in the `ParseUint` as a limit later. Will just go for 64 bits, this ain't the 80s.